utils_test.go

```diff
@@ -145,4 +146,38 @@ func testUtils(t *testing.T, when spec.G, it spec.S) {
 			}
 		})
 	})
+
+	when(".ResolveRunImage", func() {
+		when("when there are no mirrors", func() {
+			it("should return run-image", func() {
+				md := platform.StackMetadata{RunImage: platform.StackRunImageMetadata{Image: "company/run:focal"}}
+				dstImage := "someregistry/whatever"
+				res, err := lifecycle.ResolveRunImage(md, dstImage)
+				h.AssertNil(t, err)
+				h.AssertEq(t, res, md.RunImage.Image)
+			})
+		})
+
+		when("when there are mirrors", func() {
+			it("should return a run-image from the mirror matching the destination image", func() {
+				md := platform.StackMetadata{RunImage: platform.StackRunImageMetadata{
+					Image:   "company/run:focal",
+					Mirrors: []string{"some.registry/_/run:focal"},
+				}}
+				dstImage := "some.registry/app/web"
+				res, err := lifecycle.ResolveRunImage(md, dstImage)
+				h.AssertNil(t, err)
+				h.AssertEq(t, res, "some.registry/_/run:focal")
+			})
+		})
+
+		when("when there is no run image defined", func() {
+			it("should fail", func() {
+				md := platform.StackMetadata{}
+				dstImage := "someregistry/whatever"
+				_, err := lifecycle.ResolveRunImage(md, dstImage)
+				h.AssertError(t, err, "CNB_RUN_IMAGE is required when there is no stack metadata available")
```

Comment:
Could we add a test that uses `CNB_RUN_IMAGE`?

Reply:
Actually, no. The `FlagRunImage` is what ends up reading from `CNB_RUN_IMAGE ` to set `runImage`. This message maybe should be updated since it comes from this function that doesn't deal with it, but this helper doesn't deal with resolving the env var override. See `analyzer.getRunImage`.

Reply:
This makes sense. In that case I think we should change the error message (to avoid confusion). 

It would be nice to advise the user how they might supply the run image when it's missing from the stack metadata (perhaps `analyzer.getRunImage` could do this) - at the same time, I wonder if `CNB_RUN_IMAGE is required` is too specific - because they could also pass `-run-image`, no? It seems like we'd want something like `run image not provided via argument, environment, or stack metadata` but that reads a little weirdly to me, not sure how to improve it...